azurelinuxagent/common/protocol/wire.py

```diff
@@ -628,7 +628,7 @@ def stream(self, uri, destination, headers=None, use_proxy=None):
         logger.verbose("Fetch [{0}] with headers [{1}] to file [{2}]", uri, headers, destination)
 
         response = self._fetch_response(uri, headers, use_proxy)
-        if response is not None:
+        if not restutil.request_failed(response):
```

Comment:
Main change of the PR. The response can be "404: not found" which is not `None`, so according to the old condition, we would continue parsing the data here.